connector/avro/src/main/scala/org/apache/spark/sql/v2/avro/AvroDataSourceV2.scala

```diff
@@ -16,8 +16,8 @@
  */
 package org.apache.spark.sql.v2.avro
 
-import org.apache.spark.sql.avro.AvroFileFormat
 import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.core.avro.AvroFileFormat
```

Comment:
@ericm-db could you explain why the renaming is necessary? Is there any way to work around this without changing the package name?

Reply:
Can we use a Plugin or Service Loader Mechanism approach? E.g we can create a trait in `sql/core` and implement it Avro data source.

Reply:
@gengliangwang Wouldn't we still need to 'pull' the impl into `sql/core`?

Reply:
ok, we can consider moving internal implementations and keep the existing public interfaces. See my other comments in this PR.